### DISCUSSION TITLE
My cs50 codespaces suddenly dissappear

### DISCUSSION THREAD

**Discussion by Reinzkarnize** [community/community#48542] [Codespaces]
### Select Topic Area

Question

### Body

I was using my CS50 Codespaces and committing files during the night. However, when I tried to access my Codespaces in the morning, it had disappeared. Fortunately, I could still see my CS50 repository with all the commits. What steps can I take to recover my CS50 Codespaces?

**Comment by maduinya** [comment]
I have the same exact problem....  Did you get any response from the GitHub Team?

**Reply by Reinzkarnize** [reply]
Not yet, thankfully i can still use codespace using cs50 url

**Reply by larissa1500** [reply]
I have the same problem. How do you connect with the cs50 url?

**Reply by Wet-Senker** [reply]
I have the same problem. It is there in github, but Visual Studio can't reach it. 

Any solution yet? Thanks in advance.

**Reply by Reinzkarnize** [reply]
> I have the same problem. How do you connect with the cs50 url?

I navigate to [code.cs50.io](https://code.cs50.io/), where I am prompted to log in. Once logged in, It will reopen my most recent work and create a new codespace

**Reply by larissa1500** [reply]
Thanks, this was what worked for me too.

**Reply by Tcoder2022** [reply]
Thanks, helped me too. I had the same situation: the cs50 codespace had disappeared, but repository was there.

**Comment by maduinya** [comment]
Thanks for responding, Reinzkarnize. I resorted to using the VScode installed on my desktop, but I still need to upload my work. How does one even open a case for an issue such as this? I searched in vain for a tool but found none.

**Comment by larissa1500** [comment]
When connecting with the link on this page https://code.cs50.io/ , I got a new codespace. I think the files are there but the executables are gone. There is a message on the page regarding recent problems with an email contact.

**Reply by maduinya** [reply]
Thanks for responding: [ https://code.cs50.io/  ] works. Our storage have been stripped of multimedia files. So, backup your work, then rebuild.
(1) change directory to your home directory by simply typing:  cd
(2) create a temporary directory (lets say feb25tempo):  [ mkdir feb25tempo ]  <=ignore the square brackets
(3) cd to the directory you are currently working on [ cd recover ]
(4) move all your work away [ mv *.* ../feb25tempo ]
(5) run wget to rebuild the dir (recover) that you earlier backed up (accept ALL), remove recover.unzip (you don't have to... not an impediment)
(6) cd into recover and copy over your source code [ cd recover ] followed by [ cp ../feb25tempo/recover.c . ]  <= The dots matter
Thanks

**Reply by larissa1500** [reply]
Thank you for your reply. My new codespace works fine, all the .c files are there, so I can work with it as it is.

**Comment by dmilburn** [comment]
hi from the codespaces team,

over the weekend, the cs50 organization changed their GitHub plan by mistake which created some cascading changes, such as some codespaces getting deleted. we're sorry for the inconvenience. 

we've been able to restore almost all of the codespaces that were deleted. some students had already pushed all of their changes from their codespace to github. in that case, we won't be restoring the codespace directly since a new one can be created without any code loss.

If you continue to see any files missing, please go to https://code.cs50.io/ which has instructions you can follow to find any missing work


![Screenshot 2023-02-27 at 5 19 46 PM](https://user-images.githubusercontent.com/10053402/221698949-2ab3ad6f-2f87-441f-987f-62262a27460a.png)

**Comment by tandra02** [comment]
For a long time I haven't used my codespace. Now I cant access my old codespace as it has been deleted. What should I do? How can I recover it?

**Comment by CaptainHaddokk** [comment]
Similar problem. What should I do? Strangely, I log in and have my solutions to PS0 but not the rest of the problem sets (I solved all CS50Python PSs)

**Comment by Pikksel** [comment]
Same problem! Starting week 7 and have been coding every week, still, it got deleted? (Some wrote there´s a 30 day period if not used).
All my earlier code is on Github, so I could start a new workspace and continue, but now I need to install Puthon add-ons all of a sudden... Is there a way to reset the workspace?
